files/src/main/scala/io/iteratee/files/EffectCapture.scala

```diff
@@ -0,0 +1,35 @@
+package io.iteratee.files
+
+import cats.ApplicativeError
+import cats.data.Xor
+import scala.util.Try
+import scala.util.control.NonFatal
+
+trait EffectCapture[F[_]] {
+  def apply[A](a: => A): F[A]
```

Comment:
@johnynek Agreed, but I think the same objections apply to the `captureEffect` in #106. In any case I'm convinced this type class experiment isn't a good idea and will probably close this PR this morning.